### PR TITLE
rz_core_task_join: Return fine-grained error values

### PR DESCRIPTION
**Pull request by kazarmy**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [X] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This pr makes `rz_core_task_join()` return fine-grained error values.

**Test plan**

All builds except the woodpecker build are green.

**Closing issues**

...

### CODE REVIEW

File: librz/include/rz_core.h [hunk 1]
```diff
@@ -1138,6 +1138,14 @@ typedef enum {
 	RZ_CORE_TASK_STATE_DONE
 } RzTaskState;
 
+typedef enum {
+	RZ_CORE_TASK_JOIN_ERR_SUCCESS = 0,
+	RZ_CORE_TASK_JOIN_ERR_CURRENT,
+	RZ_CORE_TASK_JOIN_ERR_NO_TASK,
+	RZ_CORE_TASK_JOIN_ERR_NO_SEM_SINGLE,
```
Comment (XVilka) [inline librz/include/rz_core.h]: Please add Doxygen

**Review by XVilka — APPROVED** [review]

File: librz/include/rz_core.h [hunk 2]
```diff
@@ -1138,6 +1138,14 @@ typedef enum {
...
+	RZ_CORE_TASK_JOIN_ERR_NO_SEM_GROUP
+} RzCoreTaskJoinErr;
```
Comment (wargio) [inline librz/include/rz_core.h:1147]: add a comment and then `RZ_CORE_TASK_JOIN_ERR_SIZE`.

Comment (wargio) [inline librz/include/rz_core.h]: maybe add a command to debug this?

Comment (kazarmy) [reply librz/include/rz_core.h]: example?

File: librz/include/rz_core.h (hunk 1 above)
Comment (kazarmy) [reply librz/include/rz_core.h]: ok 7d36b91e68c7ae1ebece73f3edfe686664f8cf21

**Review by XVilka — APPROVED** [review]

File: librz/include/rz_core.h (hunk 2 above)
Comment (wargio) [reply librz/include/rz_core.h]: i mean, it gets returned but it is not used anywhere. maybe is worth logging it via RZ_LOG_DEBUG or similar.

Comment (kazarmy) [reply librz/include/rz_core.h]: i agree that logging this can be useful ... currently, this is a symptom and not the cause though

**Pull request merged**